torch_xla/csrc/ops/prod.cpp

```diff
@@ -0,0 +1,71 @@
+#include "torch_xla/csrc/ops/prod.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/convert_ops.h"
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/reduction.h"
+#include "torch_xla/csrc/tensor_util.h"
+#include "torch_xla/csrc/torch_util.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::XlaOp LowerProd(const xla::XlaOp& input,
+                    const std::vector<xla::int64>& dimensions,
+                    bool keep_reduced_dimensions,
+                    c10::optional<at::ScalarType> dtype) {
+  xla::XlaOp casted_input;
+  casted_input = dtype ? ConvertTo(input, XlaHelpers::TypeOfXlaOp(input),
```

Comment:
Like in sum.cpp, is a bit more readable. When you have ternaries which spans multiple lines, better and explicit "if".

Reply:
Thank you for the suggestion.